HerPublicWebsite.BusinessLogic/ExternalServices/EpbEpc/EpbEpcApi.cs

```diff
@@ -43,38 +45,94 @@ public record struct EpcAssessment
         public EpcRating CurrentBand { get; set; }
     }
 
-    public record struct Data
+    public record class EpcDataDto
     {
         [JsonProperty("assessment")]
-        public EpcAssessment Assessment { get; set; }
+        public EpcAssessmentDto Assessment { get; set; }
     }
 
-    public record struct Root
+    public record class EpbEpcDto
     {
         [JsonProperty("data", Required = Required.Always)]
-        public Data Data { get; set; }
+        public EpcDataDto Data { get; set; }
     }
 
     public class EpbEpcApi : IEpcApi
     {
         private readonly EpbEpcConfiguration config;
+        private readonly IMemoryCache memoryCache;
+        private readonly ILogger logger;
+        private readonly string cacheTokenKey = "EpbEpcToken";
 
-        public EpbEpcApi(IOptions<EpbEpcConfiguration> options)
+        public EpbEpcApi(IOptions<EpbEpcConfiguration> options, IMemoryCache memoryCache, ILogger<EpbEpcApi> logger)
         {
-            config = options.Value;
+            this.config = options.Value;
+            this.memoryCache = memoryCache;
+            this.logger = logger;
         }
 
-        public async Task<EpcAssessment> EpcFromUprn(string uprn)
+        public async Task<EpcAssessmentDto> EpcFromUprn(string uprn)
```

Comment:
Please add some error handling around the API call (e.g. like GetEpcForId in https://github.com/UKGovernmentBEIS/beis-simple-energy-advice-beta/blob/main/SeaPublicWebsite.BusinessLogic/ExternalServices/EpbEpc/EpbEpcApi.cs)
We don't want the entire page to submission to fail if the EPC API is down, we can just treat it the same as not finding an EPC.

Reply:
In fact GetEpcForId could be improved by putting the call to RequestTokenIfNeeded inside the try/catch too